minefield_game/statePattern/GameContext.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include <memory>
+#include "../board_header.h"
+#include "../player_header.h"
+
+class GameState;
+
+class GameContext
+{
+private:
+    std::unique_ptr<GameState> currentState;
```

Comment:
#nmi why making this `private` and everything else `public`? Moreso considering that there seem to be no invariance for this field to justify the encapsulation - aside from it, would look like the `GameContext` could have been a `struct` instead (pretty much as in [the cell case](https://github.com/ornella-mansilla/Minefield_Repository/pull/8/files#r2207916592)).